pyanaconda/core/util.py

```diff
@@ -1490,3 +1490,30 @@ def __setattr__(self, name, value):
             return super().__setattr__(name, value)
 
         return setattr(self._object, name, value)
+
+
+_os_version_finder = re.compile(r"VERSION_ID=[\"']?([0-9a-z._-]+)[\"']?")
+
+
+def get_os_version(sysroot=""):
+    """Find version of the OS from the os-release file.
+
+    See os-release(5).
+
+    :param str sysroot: Where to look.
+    :return str: The version
+    """
+    with open(sysroot + "/etc/os-release", "r") as f:
+        data = f.read(4096)  # 4 kB should be enough for everyone!
+    try:
+        return _os_version_finder.findall(data)[0]
+    except IndexError:
+        pass
+
+    with open(sysroot + "/usr/lib/os-release", "r") as f:
+        data = f.read(4096)
+    try:
+        return _os_version_finder.findall(data)[0]
+    except IndexError:
+        pass
+    return None
```

Comment:
This might be a better way to actually _read_ `os-release(5)` data: https://pagure.io/fedora-rust/rust2rpm/blob/master/f/rust2rpm/__main__.py#_37-44